.github/actions/action-npm-build/action.yml

```
@@ -0,0 +1,39 @@
+name: Compliance check and build test
+
+inputs:
+  node-version:
+    required: false
+    default: 20
+    type: string
+  app-path:
+    required: true
+    type: string
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v4
+    - name: Set up Node.js
+      uses: actions/setup-node@v4
+      with:
+        node-version: ${{ inputs.node-version }}
+        cache: "npm"
+        cache-dependency-path: "./${{inputs.app-path}}/package-lock.json"
+    - name: Install dependencies
+      run: npm --prefix ./${{inputs.app-path}} ci
+      shell: bash
+    - name: Run lint
+      run: npm run --prefix  ./${{inputs.app-path}} lint
+      shell: bash
+    - name: Run test
+      run: npm run  --prefix ./${{inputs.app-path}} test:unit
+      shell: bash
+    - name: Run build
+      run: npm run  --prefix ./${{inputs.app-path}} build
+      shell: bash
+    - name: "Upload Artifact"
+      uses: actions/upload-artifact@v4
+      with:
+        name: ${{inputs.app-path}}
```

```suggestion
        name: "${{inputs.app-path}}-node"
```

Probably needed to not collision with maven build

---

glaube nicht. https://github.com/it-at-m/sps/pull/11/files#r1751265182

Alte refarch-Projekte werden nicht mehr unterstützt. Daher hat jeder Pfad entweder ein Node Projekt oder ein Openjdk Proejkt

---

Makes sense 👍